process/rating/ratingsData.go

```diff
@@ -96,15 +106,74 @@ func NewRatingsData(args RatingsDataArg) (*RatingsData, error) {
 		return nil, err
 	}
 
-	return &RatingsData{
-		startRating:           ratingsConfig.General.StartRating,
-		maxRating:             ratingsConfig.General.MaxRating,
-		minRating:             ratingsConfig.General.MinRating,
-		signedBlocksThreshold: ratingsConfig.General.SignedBlocksThreshold,
-		metaRatingsStepData:   metaRatingStep,
-		shardRatingsStepData:  shardRatingStep,
-		selectionChances:      chances,
-	}, nil
+	ratingData := &RatingsData{
+		startRating:                 ratingsConfig.General.StartRating,
+		maxRating:                   ratingsConfig.General.MaxRating,
+		minRating:                   ratingsConfig.General.MinRating,
+		signedBlocksThreshold:       ratingsConfig.General.SignedBlocksThreshold,
+		metaRatingsStepData:         metaRatingStep,
+		shardRatingsStepData:        shardRatingStep,
+		selectionChances:            chances,
+		nodesSetup:                  args.NodesSetupHandler,
+		ratingsSetup:                ratingsConfig,
+		roundDurationInMilliseconds: args.RoundDurationMilliseconds,
+	}
+
+	args.EpochNotifier.RegisterNotifyHandler(ratingData)
+
+	return ratingData, nil
+}
+
+// EpochConfirmed will be called whenever a new epoch is called
+func (rd *RatingsData) EpochConfirmed(epoch uint32, _ uint64) {
+	// TODO: maybe check if new data is the same and avoid re-computing (old consensus group size = new consensus group size, and so on)
+	// analyze if epoch subscription is done right, since this function must be called after the nodesSetupHandler's one
+	log.Debug("RatingsData - epoch confirmed", "epoch", epoch)
+
+	rd.mutConfiguration.Lock()
+	defer rd.mutConfiguration.Unlock()
+
+	shardRatingsStepData, err := computeRatingStep(computeRatingStepArg{
+		shardSize:                       rd.nodesSetup.MinNumberOfShardNodes(),
```

Comment:
can we create a function for generating the args for Metachain and another one for Shard?
then use that method on both constructor and here?